src/lib.rs

```diff
@@ -4091,6 +4091,217 @@ pub fn digraph_random_layout(
+///
+/// :param PyDiGraph graph: The graph to generate the layout for
+/// :param list nlist: The list of lists of indexes which represents each shell
+/// :param float rotate: Angle by which to rotate the starting position of each shell
+///     relative to the starting position of the previous shell (in radians)
+/// :param float scale: An optional scaling factor to scale positions
+/// :param tuple center: An optional center position. This is a 2 tuple of two
+///     ``float`` values for the center position
+///
+/// :returns: The shell layout of the graph.
+/// :rtype: Pos2DMapping
+#[pyfunction]
+#[text_signature = "(graph, /, nlist=None, rotate=None, scale=1, center=None)"]
+pub fn digraph_shell_layout(
+    graph: &digraph::PyDiGraph,
+    nlist: Option<Vec<Vec<usize>>>,
+    rotate: Option<f64>,
+    scale: Option<f64>,
+    center: Option<layout::Point>,
+) -> Pos2DMapping {
+    layout::shell_layout(&graph.graph, nlist, rotate, scale, center)
+}
+
+/// Generate a spiral layout of the graph
+///
+/// :param PyGraph graph: The graph to generate the layout for
+/// :param float scale: An optional scaling factor to scale positions
+/// :param tuple center: An optional center position. This is a 2 tuple of two
+///     ``float`` values for the center position
+/// :param float resolution: The compactness of the spiral layout returned.
+///     Lower values result in more compressed spiral layouts.
+/// :param bool equidistant: If true, nodes will be plotted equidistant from
+///     each other.
+///
+/// :returns: The spiral layout of the graph.
+/// :rtype: Pos2DMapping
+#[pyfunction]
+#[text_signature = "(graph, /, scale=1, center=None, resolution=0.35, 
+                     equidistant=False)"]
+pub fn graph_spiral_layout(
+    graph: &graph::PyGraph,
+    scale: Option<f64>,
+    center: Option<layout::Point>,
+    resolution: Option<f64>,
+    equidistant: Option<bool>,
+) -> Pos2DMapping {
+    layout::spiral_layout(&graph.graph, scale, center, resolution, equidistant)
+}
+
+/// Generate a spiral layout of the graph
+///
+/// :param PyDiGraph graph: The graph to generate the layout for
+/// :param float scale: An optional scaling factor to scale positions
+/// :param tuple center: An optional center position. This is a 2 tuple of two
+///     ``float`` values for the center position
+/// :param float resolution: The compactness of the spiral layout returned.
+///     Lower values result in more compressed spiral layouts.
+/// :param bool equidistant: If true, nodes will be plotted equidistant from
+///     each other.
+///
+/// :returns: The spiral layout of the graph.
+/// :rtype: Pos2DMapping
+#[pyfunction]
+#[text_signature = "(graph, /, scale=1, center=None, resolution=0.35, 
+                     equidistant=False)"]
+pub fn digraph_spiral_layout(
+    graph: &digraph::PyDiGraph,
+    scale: Option<f64>,
+    center: Option<layout::Point>,
+    resolution: Option<f64>,
+    equidistant: Option<bool>,
+) -> Pos2DMapping {
```

Comment:
Personally I would do something like:

```suggestion
#[pyfunction(scale="1.0", resolution="0.35", equidistant="false")]
#[text_signature = "(graph, /, scale=1, center=None, resolution=0.35, 
                     equidistant=False)"]
pub fn digraph_spiral_layout(
    graph: &digraph::PyDiGraph,
    scale: f64,
    center: Option<layout::Point>,
    resolution: f64,
    equidistant: bool,
) -> Pos2DMapping {
```

so we set the default in the macro generated python c api ffi function. But this also works fine